src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -0,0 +1,483 @@
+                        reducedMethodSymbol = methodSymbol.ReduceExtensionMethod(receiverTypeSymbol);
+                    }
+
+                    if (reducedMethodSymbol != null)
+                    {
+                        var symbolKeyData = SymbolKey.CreateString(reducedMethodSymbol);
+                        builder.Add(new SerializableImportCompletionItem(
+                            symbolKeyData,
+                            reducedMethodSymbol.Name,
+                            reducedMethodSymbol.Arity,
+                            reducedMethodSymbol.GetGlyph(),
+                            qualifiedNamespaceName));
+                    }
+                }
+            }
+        }
+
+        private static void GetItemsFromConflictingTypes(
+            INamespaceSymbol containingNamespaceSymbol,
+            ConflictNameNode conflictTypeNodes,
+            ArrayBuilder<SerializableImportCompletionItem> builder,
+            ITypeSymbol receiverTypeSymbol,
+            SemanticModel semanticModel,
+            int position,
+            StatisticCounter counter)
+        {
+            Debug.Assert(!conflictTypeNodes.NamespaceAndMethodNames.HasValue);
+
+            foreach (var child in conflictTypeNodes.Children.Values)
+            {
+                if (child.NamespaceAndMethodNames == null)
+                {
+                    var childNamespace = containingNamespaceSymbol.GetMembers(child.Name).OfType<INamespaceSymbol>().FirstOrDefault();
+                    if (childNamespace != null)
+                    {
+                        GetItemsFromConflictingTypes(childNamespace, child, builder, receiverTypeSymbol, semanticModel, position, counter);
+                    }
+                }
+                else
+                {
+                    var types = containingNamespaceSymbol.GetMembers(child.Name).OfType<INamedTypeSymbol>();
+                    foreach (var type in types)
+                    {
+                        var (namespaceName, methodNames) = child.NamespaceAndMethodNames.Value;
+                        GetItemsFromTypeContainsPotentialMatches(type, namespaceName, methodNames, receiverTypeSymbol, semanticModel, position, counter, builder);
+                    }
+                }
+            }
+        }
+
+        // We only call this when the containing symbol is accessible, 
+        // so being declared as public means this symbol is also accessible.
+        private static bool IsSymbolAccessible(ISymbol symbol, int position, SemanticModel semanticModel)
+            => symbol.DeclaredAccessibility == Accessibility.Public || semanticModel.IsAccessible(position, symbol);
+
+        /// <summary>
+        /// The purpose of this is to help us keeping track of conflicting types with data required to create corresponding
+        /// items in a tree, which is easy to use while navigating symbol tree recursively.
+        /// </summary>
+        private class ConflictNameNode : IDisposable
```

Comment:
i'm not really sure how conflicts arise.  your indices already are associated with source or pe-references.  so you can triviall walk the corresponding IAssembly for each to find your matches, and you should never run into a conflict in these specific iassembly symbols.